### ISSUE TITLE
Impossible to reload/restart a new Centreon Engine poller from Centreon central.

### ISSUE THREAD

**Issue by mickael-decastro** [centreon/centreon#5869]
---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------

**Centreon Web version**:  2.8.15

**Centreon Engine version**: 1.8.0

**Centreon Broker version**: 3.0

**OS**: CentOS 6

**Additional environment details (AWS, VirtualBox, physical, etc.):** Virtual Machine on Vmware infratructure and OVH public cloud.

**Steps to reproduce the issue:**
1. Add new poller en centreon 
2. Add host to the new poller
3. Reload poller configuration
4. Open log /var/log/centreon/centcore.log

**Describe the results you received:**
The configuration files are not generated on the central server. Folder 
On the centcore.log we have ce following result. 
2017-11-08 18:01:09 - Engine configuration file is empty for poller 24. Please check nagios.cfg file.
2017-11-08 18:01:09 - Cannot reload Engine for poller 24
2017-11-08 18:01:09 - Init Script : 'sudo service  reload' On poller  (24)
2017-11-08 18:01:09 - Init Script : 'sudo service  reload' On poller  (24)

**Describe the results you expected:**
The configuration files should be transfered on the new poller et centengine service restarted. 
When everything is ok we have ce following result on centcore.log. 
2017-11-08 18:07:46 - Start: Send config files on poller 12
2017-11-08 18:07:49 - Result :
2017-11-08 18:07:49 - End: Send config files on poller 12
2017-11-08 18:07:49 - Start: Send Centreon Broker config files on poller 12
2017-11-08 18:07:50 - Result :
2017-11-08 18:07:50 - End: Send Centreon Broker config files on poller 12
2017-11-08 18:07:50 - End: Send Centreon Broker config files on poller 12
2017-11-08 18:07:51 - Init Script : 'sudo service centengine reload' On poller 90.85.168.168 (12)
2017-11-08 18:07:51 - Engine : centengine reload: done[  OK  ]
2017-11-08 18:07:51 - Engine : centengine reload: done[  OK  ]

**Additional information you think important (e.g. issue happens only occasionally):**

**Comment by lpinsivy** [comment]
Hi,

What is the OS of the poller?

The file /etc/sudoers.d/centreon is defined?
SELinux is disabled?

**Comment by mickael-decastro** [comment]
Hi Laurent,



The poller OS is Centos 6



Yes, the file /etc/sudoers.d/centreon is defined like that



*# cat /etc/sudoers.d/centreon*

*## BEGIN: CENTREON SUDO*



*User_Alias      CENTREON=%centreon*

*Defaults:CENTREON !requiretty*



*# centreontrapd*

*CENTREON   ALL = NOPASSWD: /sbin/service centreontrapd start*

*CENTREON   ALL = NOPASSWD: /sbin/service centreontrapd stop*

*CENTREON   ALL = NOPASSWD: /sbin/service centreontrapd restart*

*CENTREON   ALL = NOPASSWD: /sbin/service centreontrapd reload*

*CENTREON   ALL = NOPASSWD: /usr/sbin/service centreontrapd start*

*CENTREON   ALL = NOPASSWD: /usr/sbin/service centreontrapd stop*

*CENTREON   ALL = NOPASSWD: /usr/sbin/service centreontrapd restart*

*CENTREON   ALL = NOPASSWD: /usr/sbin/service centreontrapd reload*



*# Centreon Engine*

*CENTREON   ALL = NOPASSWD: /sbin/service centengine start*

*CENTREON   ALL = NOPASSWD: /sbin/service centengine stop*

*CENTREON   ALL = NOPASSWD: /sbin/service centengine restart*

*CENTREON   ALL = NOPASSWD: /sbin/service centengine reload*

*CENTREON   ALL = NOPASSWD: /usr/sbin/service centengine start*

*CENTREON   ALL = NOPASSWD: /usr/sbin/service centengine stop*

*CENTREON   ALL = NOPASSWD: /usr/sbin/service centengine restart*

*CENTREON   ALL = NOPASSWD: /usr/sbin/service centengine reload*



*# Centreon Broker*

*CENTREON   ALL = NOPASSWD: /sbin/service cbd start*

*CENTREON   ALL = NOPASSWD: /sbin/service cbd stop*

*CENTREON   ALL = NOPASSWD: /sbin/service cbd restart*

*CENTREON   ALL = NOPASSWD: /sbin/service cbd reload*

*CENTREON   ALL = NOPASSWD: /usr/sbin/service cbd start*

*CENTREON   ALL = NOPASSWD: /usr/sbin/service cbd stop*

*CENTREON   ALL = NOPASSWD: /usr/sbin/service cbd restart*

*CENTREON   ALL = NOPASSWD: /usr/sbin/service cbd reload*



*## END: CENTREON SUDO*





Yes, SELINUX is disabled.



*Cordialement, Best regards,*

*Mickaël De Castro*

Tel : +33(0)9.72.29.05.50

Port : +33(0)6.35.45.74.06

mailto : mickael.decastro@cpf-informatique.fr

*CPF-INFORMATIQUE* (www.cpf-informatique.fr)

249, route de Pégomas

Immeuble NEOPARFUMS

06130 GRASSE



*From:* Laurent Pinsivy [mailto:notifications@github.com]
*Sent:* vendredi 10 novembre 2017 13:59
*To:* centreon/centreon <centreon@noreply.github.com>
*Cc:* mickael-decastro <mickael.decastro@cpf-informatique.fr>; Author <
author@noreply.github.com>
*Subject:* Re: [centreon/centreon] Impossible to reload/restart a new
Centreon Engine poller from Centreon central. (#5869)



Hi,

What is the OS of the poller?

The file /etc/sudoers.d/centreon is defined?
SELinux is disabled?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/centreon/centreon/issues/5869#issuecomment-343466941>,
or mute the thread

**Comment by MyNameIsAlvar** [comment]
i'm having similar issue using Centreon 2.8.15 (fresh install).
Steps to reproduce:

using CentOS-7-x86_64-Minimal-1708.iso
selinux disable, yum update && reboot

[root@centreon1 ~]# cat /etc/centos-release
CentOS Linux release 7.4.1708 (Core)

yum install net-tools.x86_64 vim wget
systemctl stop firewalld

from https://documentation.centreon.com/docs/centreon/en/latest/installation/from_packages.html#repository-install

wget http://yum.centreon.com/standard/3.4/el7/stable/noarch/RPMS/centreon-release-3.4-4.el7.centos.noarch.rpm
yum install --nogpgcheck centreon-release-3.4-4.el7.centos.noarch.rpm
yum install centreon-base-config-centreon-engine centreon MariaDB-server

echo "date.timezone = Europe/Tallinn"  >> /etc/php.d/php-timezone.ini 

mkdir -p  /etc/systemd/system/mariadb.service.d/
echo -ne "[Service]\nLimitNOFILE=32000\n" | tee /etc/systemd/system/mariadb.service.d/limits.conf
systemctl daemon-reload
service mysql restart

systemctl enable httpd.service
systemctl enable snmpd.service
systemctl enable mysql.service
systemctl start httpd snmpd

-----------------------------
after clicking through the installation, i define a host, service and check command for both.
Then Configuration -> pollers -> export configuration -> select my poller, 4x ticks to action boxes 
 && Method restart -> Export

output:
Preparing environment... OK
Generating files... OK
Moving files... OK
Restarting engine... OK

NB - pid of centengine process chanes every time.
[root@centreon1 ~]# pidof centengine
13486


But "pollers" view still shows "NO" under "Is running?"
What i'm doing wrong?

**Comment by srubioco** [comment]
Hi,

I'm using Centos 6.9
Centreon Web version: 2.8.15
Centreon Engine version: 1.8.0.2
Centreon Broker version: 3.0

For me the remote poller is running correctly, but not restart the services from web. Only apply the configuration when I restart centengine manually on the remote poller.

On centcore.log I see this problem:

[root@Centreon-Central centreon]# tail -f centcore.log
2017-11-20 10:33:31 - Unknown init system for poller 5

Is reproduced when I launch by CLI or via Web:

[root@Centreon-Central 1]# centreon -u admin -p centreon -a POLLERTEST -v 5
Warning: Centreon Poller 5 can restart but configuration is not optimal. Please see debug bellow :
---------------------------------------------------------------------------------------------------
[1511179668] [31679] Reading main configuration file '/usr/share/centreon//filesGeneration/engine/5/centengine.DEBUG'.
[1511179668] [31679] Processing object config file '/usr/share/centreon//filesGeneration/engine/5/hostTemplates.cfg'
[1511179668] [31679] Processing object config file '/usr/share/centreon//filesGeneration/engine/5/hosts.cfg'
[1511179668] [31679] Processing object config file '/usr/share/centreon//filesGeneration/engine/5/serviceTemplates.cfg'
[1511179668] [31679] Processing object config file '/usr/share/centreon//filesGeneration/engine/5/services.cfg'
[1511179668] [31679] Processing object config file '/usr/share/centreon//filesGeneration/engine/5/commands.cfg'
[1511179668] [31679] Processing object config file '/usr/share/centreon//filesGeneration/engine/5/contactgroups.cfg'
[1511179668] [31679] Processing object config file '/usr/share/centreon//filesGeneration/engine/5/contacts.cfg'
[1511179668] [31679] Processing object config file '/usr/share/centreon//filesGeneration/engine/5/hostgroups.cfg'
[1511179668] [31679] Processing object config file '/usr/share/centreon//filesGeneration/engine/5/servicegroups.cfg'
[1511179668] [31679] Processing object config file '/usr/share/centreon//filesGeneration/engine/5/timeperiods.cfg'
[1511179668] [31679] Processing object config file '/usr/share/centreon//filesGeneration/engine/5/escalations.cfg'
[1511179668] [31679] Processing object config file '/usr/share/centreon//filesGeneration/engine/5/dependencies.cfg'
[1511179668] [31679] Processing object config file '/usr/share/centreon//filesGeneration/engine/5/connectors.cfg'
[1511179668] [31679] Processing object config file '/usr/share/centreon//filesGeneration/engine/5/meta_commands.cfg'
[1511179668] [31679] Processing object config file '/usr/share/centreon//filesGeneration/engine/5/meta_timeperiod.cfg'
[1511179668] [31679] Processing object config file '/usr/share/centreon//filesGeneration/engine/5/meta_host.cfg'
[1511179668] [31679] Processing object config file '/usr/share/centreon//filesGeneration/engine/5/meta_services.cfg'
[1511179668] [31679] Reading resource file '/usr/share/centreon//filesGeneration/engine/5/resource.cfg'
[1511179668] [31679] Warning: Service 'Ping' on host 'cctv01_poller' has no notification time period defined!
[1511179668] [31679] Checking global event handlers...
[1511179668] [31679] Checking obsessive compulsive processor commands...
[1511179668] [31679]
[1511179668] [31679] Checked 2 commands.
[1511179668] [31679] Checked 0 connectors.
[1511179668] [31679] Checked 0 contacts.
[1511179668] [31679] Checked 0 host dependencies.
[1511179668] [31679] Checked 0 host escalations.
[1511179668] [31679] Checked 0 host groups.
[1511179668] [31679] Checked 1 hosts.
[1511179668] [31679] Checked 0 service dependencies.
[1511179668] [31679] Checked 0 service escalations.
[1511179668] [31679] Checked 0 service groups.
[1511179668] [31679] Checked 1 services.
[1511179668] [31679] Checked 1 time periods.
[1511179668] [31679]
[1511179668] [31679] Total Warnings: 1
[1511179668] [31679] Total Errors:   0

---------------------------------------------------------------------------------------------------
Return code end : 0

**Comment by lpinsivy** [comment]
@MyNameIsAlvar can you execute:

 select id, name, init_script, init_system from nagios_server;
+----+---------+-------------+-------------+
| id | name    | init_script | init_system |
+----+---------+-------------+-------------+
|  1 | Central | centengine  | systemv     |
|  9 | Poller  | centengine  | systemv        |
+----+---------+-------------+-------------+

**Comment by MyNameIsAlvar** [comment]
MariaDB [centreon]> select id, name, init_script, init_system from nagios_server;
+----+---------+-------------+-------------+
| id | name    | init_script | init_system |
+----+---------+-------------+-------------+
|  1 | Central | centengine  | NULL        |
+----+---------+-------------+-------------+
1 row in set (0.00 sec)

**Comment by srubioco** [comment]
In my case:
MariaDB [centreon]> select id, name, init_script, init_system from nagios_server;
+----+-----------------+-------------+-------------+
| id | name            | init_script | init_system |
+----+-----------------+-------------+-------------+
|  1 | Central         | centengine  | systemv     |
|  2 | Poller-template | centengine  | systemv     |
|  5 | Poller-aytoval  | centengine  | NULL        |
+----+-----------------+-------------+-------------+
3 rows in set (0.00 sec)

**Comment by MyNameIsAlvar** [comment]
Fore some reason.. restarting the whole server changes someting. Sql output stays the same, but now GUI is showing that poller is running. What changes? What component needs to be restarted? (noticed it after shutting the test VM fot the night and powering it up for now).
I'm kind of unable to restart the whole server in production environment, ofcourse...

So i have currently 2 identical installations on my hands - one restarted and other not. Ask whatever conf/parameters you like.

**Comment by srubioco** [comment]
I solve my problem, but I'm not sure if it's permanent.

First I edited the database to set the init_system and monitoring_engine that was nulls before:

UPDATE nagios_server SET init_system = 'systemv' WHERE id=5;
UPDATE nagios_server SET monitoring_engine = 'CENGINE' WHERE id=5;

MariaDB [centreon]> select id, name, init_script, init_system, monitoring_engine from nagios_server;
+----+-----------------+------------------------+-------------+-------------------+
| id | name            | init_script            | init_system | monitoring_engine |
+----+-----------------+------------------------+-------------+-------------------+
|  1 | Central         | centengine             | systemv     | CENGINE           |
|  2 | Poller-template | centengine             | systemv     | CENGINE           |
|  5 | Poller-aytoval  | centengine | systemv     | CENGINE           |
+----+-----------------+------------------------+-------------+-------------------+

After this, I try to restart services via web, with this error:
sudo: no tty present and no askpass program specified

I edited my /etc/sudoers adding this line:
CENTREON ALL = NOPASSWD: ALL

And again I try to restart the services via web, with this error:
[1511191165] [20434] Error: Parsing of global configuration failed: Can't open file '/var/spool/centreon/restart'
[1511191839] [20524] Error: Parsing of global configuration failed: Can't open file '/var/spool/centreon/reload'

Finally, I change on poller configuration the Monitoring Engine Script to: /etc/init.d/centengine

And works!!

But sometimes, when I restart the services of my Central poller I lost my first change in the database, changing automatically the init_system of my remote poller to NULL.

**Comment by MyNameIsAlvar** [comment]
update:
it seems that cbd.service fails to start in the  background:

[root@test centreon-broker]# systemctl status cbd.service
● cbd.service - Centreon Broker watchdog
   Loaded: loaded (/etc/systemd/system/cbd.service; enabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Tue 2017-11-21 17:01:33 EET; 18s ago
  Process: 549 ExecStart=/usr/sbin/cbwd /etc/centreon-broker/watchdog.xml (code=exited, status=127)
 Main PID: 549 (code=exited, status=127)

i'm trying to figure out, why and how can i fix it.

Nov 21 17:01:33 test systemd[1]: Started Centreon Broker watchdog.
Nov 21 17:01:33 test systemd[1]: Starting Centreon Broker watchdog...
Nov 21 17:01:33 test cbwd[549]: /usr/sbin/cbwd: symbol lookup error: /usr/sbin/cbwd: undefined symbol: _ZN16QCoreApplicationC1ERiPPci
Nov 21 17:01:33 test systemd[1]: cbd.service: main process exited, code=exited, status=127/n/a
Nov 21 17:01:33 test systemd[1]: Unit cbd.service entered failed state.
Nov 21 17:01:33 test systemd[1]: cbd.service failed.

**Comment by MyNameIsAlvar** [comment]
sadly all my problems were caused by some el6 packages  i mistakenly installed previously...
What i learned:

don't mix centos6 and centos7 centreon repositories (and don't forget that you did so).

[centreon@test ~]$ readelf -d /usr/sbin/cbwd|grep "Shared library"|awk -F \[ '{print "ls -la `locate ",$2,"`"}'|tr -d "]"|sh

it showed me difference with /usr/lib64/libQtXml.so.4

[root@test ~]# yum whatprovides /usr/lib64/libQtXml.so.4
lead me to qt-4.7.4-0.el6.x86_64 what was mistakenly installed and left intact.

**Comment by CoSmaConsulting** [comment]
We have on a client-server the same issue with the 'Poller', we have tried all 'work-arounds', but nothing has worked you so fare. 
It is a complete fresh centreon.iso installation Version 2.8.22
We get this error, see screenshot. 
We updated the Version on the server and latest centos 7 updates.

![screenshot from 2018-05-27 14-47-05](https://user-images.githubusercontent.com/34801269/40590935-dd69d972-61bc-11e8-8488-755c9648ea50.png)
Please advice if anyone has a suggestion.

**Comment by CoSmaConsulting** [comment]
![screenshot from 2018-05-27 14-50-32](https://user-images.githubusercontent.com/34801269/40590959-5b51570c-61bd-11e8-93df-f427f72e3d71.png)

**Comment by CoSmaConsulting** [comment]
![screenshot from 2018-05-27 14-53-21](https://user-images.githubusercontent.com/34801269/40590981-bd295844-61bd-11e8-9ac2-881d52931281.png)

**Comment by sysadminasks** [comment]
I had the same experience. My poller stopped working after adding some checks and I wonder what happened. It turns out that I had the same issue with @MyNameIsAlvar. I used the command

`readelf -d /usr/sbin/cbwd|grep "Shared library"` to list outdated files and I saw `qt-4.7.4-0.el6.x86_64` is installed. I made sure that I am not using any old repos and then I did this.

1. yum remove qt-4.7.4-0.el6.x86_64
2. yum install libQtXml.so.4 
3. yum install centreon-broker centreon-broker-core centreon-broker-cbd centreon-broker-cbmod centreon-broker-storage
4. systemctl start centengine

My poller went back to what it is after this things that I did.

Regards,
